wallet-next/src/storage.rs

```diff
@@ -80,4 +86,76 @@ impl Storage {
             result[0].bytes.as_ref().unwrap().as_slice(),
         )?)
     }
+
+    pub async fn record_block(
+        &self,
+        scan_result: ScanResult,
+        nct: &mut NoteCommitmentTree,
+    ) -> anyhow::Result<Self> {
+
+        let mut tx = self.pool.begin().await?;
+        
+        // Insert all new note records
+        for note_record in scan_result.new_notes {
+            sqlx::query(&format!(
+                r#"INSERT INTO notes (
+                        note_commitment,
+                        height_spent,
+                        height_created,
+                        diversifier,
+                        amount,
+                        asset_id,
+                        transmission_key,
+                        blinding_factor,
+                        diversifier_index,
+                        nullifier)
+                    VALUES ({:?},NULL,{:?},{:?},{:?},{:?},{:?},{:?},{:?},{:?})"#,
+                note_record.note_commitment.0.to_bytes(),
+                //height_spent is NULL
+                scan_result.height,
+                note_record.note.diversifier().0,
+                note_record.note.amount(),
+                note_record.note.asset_id().to_bytes(),
+                note_record.note.transmission_key().0,
+                note_record.note.note_blinding().to_bytes(),
+                note_record.diversifier_index.0,
+                note_record.nullifier.to_bytes()
+            ).to_owned()).execute(&mut tx).await?;
+        }
+
+        tx.commit().await?;
+
+        // Update any rows of the table with matching nullifiers to have height_spent
+        for nullifier in scan_result.spent_nullifiers {
+            let height_spent = format!("{:?}", &scan_result.height);
+            let nullifier = format!("{:?}", &nullifier.to_bytes());
+
+            let result = query!(
+                r#"UPDATE notes 
+                SET height_spent = ?
+                WHERE nullifier = ?
+                RETURNING note_commitment"#,
+                height_spent,
+                nullifier
+            )
+            .fetch_all(&self.pool)
```

Comment:
Unsure how to include this in the `note_record` insert transaction - because this query is returning the `note_commitment` for each spent nullifier that we need to forget the corresponding NCT entry, if we execute these in a batch with the note inserts, I think we'd need to refactor to return a set of the needed `note_commitment`s on transaction commit, then update the NCT table in a separate transaction - seems doable, but is it preferable to doing this part sequentially?